dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,115 @@
+from contextlib import contextmanager
+
+import requests
+
+from .slurm import SLURMCluster
+from .core import cluster_parameters, job_parameters
+from .slurm import SLURMJob, logger
+
+
+class RemoteSLURMJob(SLURMJob):
+    def __init__(self, *args, **kwargs):
+        self.api_url = kwargs.pop("api_url")
+        self.__class__.api_url = self.api_url
+        super().__init__(*args, **kwargs)
+
+    @contextmanager
+    def job_file(self):
+        """
+        Override the job_file and pass back the job_script. job_file() is used by Job.start and
+         gives output directly to SLURMJob._submit_job().
+        """
+        yield self.job_script()
+
+    async def _submit_job(self, script):
+        # https://slurm.schedmd.com/rest_api.html#slurmctldSubmitJob
+        try:
+            response = requests.post(
+                f"{self.api_url}jobs/submit", json={"script": script}
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            logger.exception("SLURMJob request failed.")
+            raise RuntimeError from e
+        else:
+            return response.json()
+
+    def _job_id_from_submit_output(self, out):
+        # out is the JSON output from _submit_job request.post
+        # See https://slurm.schedmd.com/rest_api.html#v0.0.36_job_submission_response
+        return out["job_id"]
+
+    async def close(self):
+        logger.debug("Stopping worker: %s job: %s", self.name, self.job_id)
+        # https://slurm.schedmd.com/rest_api.html#slurmctldCancelJob
+        try:
+            response = requests.delete(f"{self.api_url}job/{self.job_id}")
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            logger.exception("SLURMJob request failed.")
```

Comment:
The parent `SLURMCluster` (specifically the root object `Cluster`) has an `_log` method which would be preferable here as we store and reuse that info in other places. 

I wonder if we could pass a pointer to the cluster object to the `RemoteSLURMJob` objects so it can be used.

Reply:
You're right that's probably an improvement. However since Job currently logs to it's own logger I'm inclined to keep that behavior here and we should figure out logging in a separate PR. (Trying to keep the scope tight).